help/gory_details/_posts/2021-04-26-author-search.md

```diff
@@ -0,0 +1,103 @@
+
+
+If you are still here, let's dive in:
+
+
+### Down the Rabbit Hole
+
+Let's take one example author: `Adamczuk, Piotr Gavrilov Eugenyi`. Heraldically attuned audience will immediately pick up that the guy is probably of Polish or Russian descend, quite likely from a noble family (blue blood runs through his veins so his name may appear next to a honorary title; but who cares? Well, our parser does). 
+
+People are unable to remember all his names, not even editors in the scientific publications, and let's face it, they are of different origins themselves. So, some type his name as `Adamcuk`, confuse his name with `Peter`, pay no attention to uncle `Gavrilov` and can't decide whether to spell his name as `Evgenij`, `Eugenij`, `Eugen`, `Evzen` and so on. Fortunately, there aren't that many persons of female gender and name Piotr, but we can't be sure about that either (yet people will still claim it is crystal clear...)
+
+Anyways, it is complicated - so let's start. Somebody (fortunately) for us, remembers spelling of the surname and types:
+
+
+`author:peter adamczuk`
+
+
+Our *search* tokenizer is going to take the input in its entirety (i.e. **NO** tokenization into `peter` and `adamzuk`) and will pass the input to a python library [https://pypi.org/project/nameparser/]. Since the search engine is written in Java, in reality the library will be executed by Jython - so we are calling Java which looks like Python, but as you can see, I'm sparring you no details...
+
+The library will recognize `surname` and `first name` most of the times correctly. So `Pope John Paul the II` would loose all his tiara and become `Paul, John` -- but we don't know anything besides the characters. We are not consulting wikipedia to find out the real name of the Pope, or the author (at least not yet). And our `Peter` is recognized as the first name, and `Adamczuk` as given.
+
+Next are some boring filters: one of them normalizes the names. `Adamczuk,               P` becomes `Adamczuk, P`, and `Lao'tzu` becomes `Lao tzu,`. And we also reject any search with 6 or more names. Why? Well, because it can lead to 6^n combinations (in the worst case) in the later stages.
+
+Next comes **transliterations** -- what's that you ask? Well, English is the Lingua Franca of scientific discourse - but only for the past 2 hundred years. The dominance of US alphabet in all things electronic is also thing of the past, but humans are forever lazy (or intelligent) and when confronted with `e`, `é`, `e` .... characters, they say 'Aha! That's `e`'. All accents lost. So naturally, they want to be able to search for `jonas` and find `jonáš` as well. And so we oblige. Internally at ADS we use `UTF-8` character encoding but we don't force the English-centric writers to think in more than 26 characters of English alphabet. Also by convention, scientific publications oftentimes insist on transliterating native author names into ASCII.
```

Comment:
* "2 hundred" -> "200" or "two-hundred"
* "US alphabet" -> "the English alphabet" (the British and Australians use the same alphabet, though different spellings, but that's not what you're referring to)
* "also thing of the past" -> "also a thing of the past"